edb/server/compiler/compiler.py

```diff
@@ -1415,6 +1415,30 @@ def _compile(
         ctx: CompileContext,
         source: edgeql.Source,
     ) -> List[dbstate.QueryUnit]:
+        try:
+            return self._try_compile(ctx=ctx, source=source)
+        except errors.EdgeQLSyntaxError as e:
+
+            if isinstance(source, edgeql.NormalizedSource):
+                # try non-normalized source
+                try:
+                    original = edgeql.Source.from_string(source.text())
+                    ctx = dataclasses.replace(ctx, original)
+                    return self._try_compile(ctx=ctx, source=original)
+                except errors.EdgeQLSyntaxError as e:
+                    raise e
+                else:
+                    raise AssertionError(
```

Comment:
I'd raise `InternalServerError` and add more context: the normalized query text; the type and text of the original error; the type and text of the new error. I know this is unlikely to happen, but if it does it would help us to debug this.

Reply:
Both of the errors are in `__source__`:
```
      Traceback (most recent call last):
        File "/work/edb/server/compiler/compiler.py", line 1419, in _compile
          return self._try_compile(ctx=ctx, source=source)
        File "/work/edb/server/compiler/compiler.py", line 1453, in _try_compile
          statements = edgeql.parse_block(source)
        File "/work/edb/edgeql/parser/__init__.py", line 69, in parse_block
          return parser.parse(source)
        File "/work/edb/common/parsing.py", line 415, in parse
          raise self.get_exception(
      edb.errors.EdgeQLSyntaxError: Unexpected '('

      During handling of the above exception, another exception occurred:

      Traceback (most recent call last):
        File "/work/edb/server/compiler/compiler.py", line 1427, in _compile
          raise RuntimeError("synthetic error");
      RuntimeError: synthetic error

      During handling of the above exception, another exception occurred:

      Traceback (most recent call last):
        File "/work/edb/server/procpool/worker.py", line 75, in worker
          res = await meth(*args)
        File "/work/edb/server/compiler/compiler.py", line 1869, in compile
          return self._compile(ctx=ctx, source=source)
        File "/work/edb/server/compiler/compiler.py", line 1432, in _compile
          raise AssertionError(
      AssertionError: Normalized and non-normalized query errors differ
```